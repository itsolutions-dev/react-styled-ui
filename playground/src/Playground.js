// @flow

import React from 'react';
import { LiveProvider, LiveEditor, LiveError, LivePreview } from 'react-live';
import ThemeProvider from '../../src/ThemeProvider';
import theme from '../../src/theme';

const getCode = code => `
  <ThemeProvider theme={theme}>
    ${code}
  </ThemeProvider>
`.trim().replace(/^/, '\t');

type PlaygroundProps = {
  code: string,
  scope: Object
};

export default ({ code, scope, ...others }: PlaygroundProps) =>
  (<LiveProvider
    code={getCode(code)}
    scope={{
      ThemeProvider,
      theme,
      ...scope,
    }}
    {...others}
  >
    <LiveEditor />
    <LiveError />
    <LivePreview />
  </LiveProvider>);
