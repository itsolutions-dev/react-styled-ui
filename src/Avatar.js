// @flow

import React from 'react';
import styled, { css } from './styledComponent';
import {
  getSizeOrDefault,
} from './utils/';

const defaultSize = '40px';
const defaultRadius = '99999px';

type AvatarProps = {
  src: string,
  alt: string,
  children: any,
};

const Avatar = styled({
  component: 'div',
  props: {
    children: ({ src, alt, children }: AvatarProps) => (
      src == null ? children : <img src={src} alt={alt || src} />
    ),
  },
  style: css`    
    ${props => (props.src == null ? `
      width: ${getSizeOrDefault(props.size, defaultSize)};
      height: ${getSizeOrDefault(props.size, defaultSize)};
      border-radius: ${getSizeOrDefault(props.radius, defaultRadius)}; 
      color: ${(props.color == null ? 'rgb(255, 255, 255)' : props.color)};
      background-color: ${(props.backgroundColor == null ? 'rgb(188, 188, 188)' : props.backgroundColor)} !important;
      font-size: ${getSizeOrDefault(props.fontSize, '20px')};
      user-select: none;
      align-items: center;
      justify-content: center;
      display: inline-flex;
    ` : `
      > img {
        width: ${getSizeOrDefault(props.size, defaultSize)};
        height: ${getSizeOrDefault(props.size, defaultSize)};
        border-radius: ${getSizeOrDefault(props.radius, defaultRadius)};
        display: inline-flex;
      } 
    `)} 
  `,
});

export default Avatar;
