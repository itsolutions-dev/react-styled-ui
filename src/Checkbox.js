// @flow

import React from 'react';
import styled, { css, keyframes } from './styledComponent';
import {
  guid,
  getBorder,
  getBackgroundColor,
} from './utils/';

const shrinkBounce = keyframes`
  0%{
    transform: scale(1);
  }
  33%{    
    transform: scale(.85);
  }
  100%{
    transform: scale(1);    
  }
`;

const check = keyframes`
  0%{
    width: 0;
    height: 0;
    border-color: #fff;
    transform: translate3d(0,0,0) rotate(45deg);
  }
  33%{
    width: .2em;
    height: 0;
    transform: translate3d(0,0,0) rotate(45deg);
  }
  100%{    
    width: .2em;
    height: .5em;    
    border-color: #fff;
    transform: translate3d(0,-.5em,0) rotate(45deg);
  }
`;

const CheckboxContainer = styled({
  component: 'div',
  style: css`
    display: flex;
    flex-direction: column;
    padding: 0.5em;

    & * {
      box-sizing: border-box;
      user-select: none;
    }

    > input[type=checkbox]{ 
      visibility: hidden;
    }

    > input[type=checkbox] + label{
      position: relative;
      display: flex;
      margin: .6em 0;
      align-items: center;
      color: #fff;
      transition: color 250ms cubic-bezier(.4,.0,.23,1);
    }

    > input[type=checkbox] + label > span{
      display: flex;
      justify-content: center;
      align-items: center;
      margin-right: 1em;
      width: 20px;
      height: 20px;
      background: transparent;
      border: 2px solid ${props => getBorder(props)};
      border-radius: 2px;
      cursor: pointer;  
      transition: all 250ms cubic-bezier(.4,.0,.23,1);
    }

    > input[type=checkbox] + label:hover, input[type=checkbox]:focus + label{
      color: #fff;
    }

    > input[type=checkbox] + label:hover > span, input[type=checkbox]:focus + label > span{
      background: rgba(255,255,255,.1);
    }

    > input[type=checkbox]:checked + label > span{
      border: .6em solid ${props => getBackgroundColor(props)};
      animation: ${shrinkBounce} 200ms cubic-bezier(.4,.0,.23,1);
    }

    > input[type=checkbox]:checked + label > span:before{
      content: '';
      position: absolute;
      top: .5em;
      left: .25em;
      border-right: 3px solid transparent;
      border-bottom: 3px solid transparent;
      transform: rotate(45deg);
      transform-origin: 0% 100%;
      animation: ${check} 125ms 250ms cubic-bezier(.4,.0,.23,1) forwards;
    }
  `,
});

type CheckboxProps = {
  // eslint-disable-next-line
  id?: any,
  primary: boolean,
};

const Checkbox = (props: CheckboxProps) => {
  const { id, ...others } = props;
  const checkboxId = id === undefined ? guid() : id;
  return (
    <CheckboxContainer {...others}>
      <input type="checkbox" id={checkboxId} {...others} />
      <label htmlFor={checkboxId}>
        <span />
      </label>
    </CheckboxContainer>
  );
};

Checkbox.defaultProps = {
  primary: true,
};

export default Checkbox;
