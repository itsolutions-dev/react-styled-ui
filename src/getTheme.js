import defaultTheme from './theme';
import deepMerge from './utils/deepMerge';

const getTheme = deepMerge.bind(null, defaultTheme);

export default getTheme;
