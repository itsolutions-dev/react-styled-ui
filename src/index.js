export Absolute from './Absolute';
export Avatar from './Avatar';
export BackgroundImage from './BackgroundImage';
export Badge from './Badge';
export Blockquote from './Blockquote';
export Border from './Border';
export Button from './Button';
export ButtonCircle from './ButtonCircle';
export ButtonGroup from './ButtonGroup';
export Card from './Card';
export Checkbox from './Checkbox';
export Code from './Code';
export Container from './Container';
export Divider from './Divider';
export Fixed from './Fixed';
export Footer from './Footer';
export getTheme from './getTheme';
export Heading from './Heading';
export Heading1 from './Heading1';
export Heading2 from './Heading2';
export Heading3 from './Heading3';
export Heading4 from './Heading4';
export Heading5 from './Heading5';
export Heading6 from './Heading6';
export Image from './Image';
export InlineText from './InlineText';
export Input from './Input';
export Label from './Label';
export Link from './Link';
export List from './List';
export ListItem from './ListItem';
export Loader from './Loader';
export Message from './Message';
export Overlay from './Overlay';
export Progress from './Progress';
export Radio from './Radio';
export Relative from './Relative';
export Select from './Select';
export Slider from './Slider';
export Text from './Text';
export Textarea from './Textarea';
export Toggle from './Toggle';
export Toolbar from './Toolbar';
export Tooltip from './Tooltip';
export Truncate from './Truncate';
export ThemeProvider from './ThemeProvider';
export theme from './theme';
