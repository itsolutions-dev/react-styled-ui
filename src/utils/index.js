export deepMerge from './deepMerge';
export getOpacity from './getOpacity';
export getProperty from './getProperty';
export getHover from './getHover';
export getBackgroundColor from './getBackgroundColor';
export getTextColor from './getTextColor';
export getBorder from './getBorder';
export getHeight from './getHeight';
export hexToRgb from './hexToRgb';
export getColorWithAlpha from './getColorWithAlpha';
export getSizeOrDefault from './getSizeOrDefault';
export guid from './guid';
