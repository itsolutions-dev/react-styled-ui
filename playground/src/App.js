import React from 'react';
import styled from 'styled-components';
import Playground from './Playground';
import * as components from '../../src/';
import ThemeProvider from '../../src/ThemeProvider';

const { Heading } = components;

const Body = styled.div`
  background: #FFFFFF;
`;

const filePathToComponentName = file =>
  file.replace(/\.\/(examples\/)?/, '').replace(/\.example$/, '');

const getExamples = () => {
  const req = require.context('./examples', true, /\.example$/);
  return req.keys().map((file) => {
    const code = req(file);
    const scope = [];

    let match;
    const componentNameRegex = /<([A-Z]\w*)[\s|>]/g;
    // eslint-disable-next-line
    while ((match = componentNameRegex.exec(code)) !== null) {
      scope.push({
        name: match[1],
        component: components[match[1]],
      });
    }

    return {
      name: filePathToComponentName(file),
      scope: scope.filter(x => x.component)
        .reduce((acc, cur) => ({
          ...acc,
          [cur.name]: cur.component,
        }), {}),
      code,
    };
  });
};

export default () => (
  <ThemeProvider>
    <Body>
      <Heading size={1}>Playground</Heading>
      {
        getExamples()
          // eslint-disable-next-line
          .sort((x, y) => x.name > y.name ? 1 : -1)
          .map(x => (
            <Playground
              key={x.name}
              code={x.code}
              scope={x.scope}
              collapsableCode
            />
          ))
      }
    </Body>
  </ThemeProvider>
);
