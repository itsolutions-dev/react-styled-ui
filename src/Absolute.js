import styled, { css } from './styledComponent';
import { getSizeOrDefault } from './utils/';

const Absolute = styled({
  component: 'div',
  style: css`
    position: absolute;
    ${props => props.top !== undefined ? `top: ${getSizeOrDefault(props.top, 0)};` : ''}
    ${props => props.bottom !== undefined ? `bottom: ${getSizeOrDefault(props.bottom, 0)};` : ''}
    ${props => props.left !== undefined ? `left: ${getSizeOrDefault(props.left, 0)};` : ''}
    ${props => props.right !== undefined ? `right: ${getSizeOrDefault(props.right, 0)};` : ''}
  `,
});

export default Absolute;
