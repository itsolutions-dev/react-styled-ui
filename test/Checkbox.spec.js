import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import getTheme from '../src/getTheme';
import Checkbox from '../src/Checkbox';
import * as utils from '../src/utils/';

const theme = getTheme({
  palette: {
    primary: '#1a5a99',
    primaryBorder: '#bababa',
  },
});

describe('Checkbox', () => {
  test('should render a Checkbox', () => {
    const component = ReactTestRenderer.create(<Checkbox theme={theme} />);
    expect(component).toBeDefined();
  });

  test('should render an input with type checkbox', () => {
    const component = ReactTestRenderer.create(<Checkbox theme={theme} />).toJSON();
    expect(component.children.find(x => x.type === 'input').props.type).toEqual('checkbox');
  });

  test('should be primary by default', () => {
    const spies = [
      jest.spyOn(utils, 'getBackgroundColor'),
      jest.spyOn(utils, 'getBorder'),
    ];
    const component = ReactTestRenderer.create(<Checkbox id="id" theme={theme} />).toJSON();
    spies.forEach(spy => expect(spy).toHaveBeenCalled());
    expect(component).toMatchSnapshot();
  });

  test('should have the provided id', () => {
    const component = ReactTestRenderer.create(<Checkbox id="foo" theme={theme} />).toJSON();
    expect(component.children.find(x => x.type === 'input').props.id).toEqual('foo');
    expect(component.children.find(x => x.type === 'label').props.htmlFor).toEqual('foo');
  });

  test('should automatically generate an id', () => {
    const component = ReactTestRenderer.create(<Checkbox theme={theme} />).toJSON();
    expect(component.children.find(x => x.type === 'input').props.id).toBeDefined();
    expect(component.children.find(x => x.type === 'label').props.htmlFor).toBeDefined();
  });

  test('should pass props to the input', () => {
    const component = ReactTestRenderer.create(<Checkbox id="id" foo="bar" theme={theme} />).toJSON();
    expect(component.children.find(x => x.type === 'input').props.foo).toEqual('bar');
  });
});
