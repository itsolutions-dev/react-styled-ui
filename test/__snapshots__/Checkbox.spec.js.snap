// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should be primary by default 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0.5em;
}

.c0 * {
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 > input[type=checkbox] {
  visibility: hidden;
}

.c0 > input[type=checkbox] + label {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: .6em 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  -webkit-transition: color 250ms cubic-bezier(.4,.0,.23,1);
  transition: color 250ms cubic-bezier(.4,.0,.23,1);
}

.c0 > input[type=checkbox] + label > span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 1em;
  width: 20px;
  height: 20px;
  background: transparent;
  border: 2px solid #bababa;
  border-radius: 2px;
  cursor: pointer;
  -webkit-transition: all 250ms cubic-bezier(.4,.0,.23,1);
  transition: all 250ms cubic-bezier(.4,.0,.23,1);
}

.c0 > input[type=checkbox] + label:hover,
.c0 input[type=checkbox]:focus + label {
  color: #fff;
}

.c0 > input[type=checkbox] + label:hover > span,
.c0 input[type=checkbox]:focus + label > span {
  background: rgba(255,255,255,.1);
}

.c0 > input[type=checkbox]:checked + label > span {
  border: .6em solid #1a5a99;
  -webkit-animation: erhqVB 200ms cubic-bezier(.4,.0,.23,1);
  animation: erhqVB 200ms cubic-bezier(.4,.0,.23,1);
}

.c0 > input[type=checkbox]:checked + label > span:before {
  content: '';
  position: absolute;
  top: .5em;
  left: .25em;
  border-right: 3px solid transparent;
  border-bottom: 3px solid transparent;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-transform-origin: 0% 100%;
  -ms-transform-origin: 0% 100%;
  transform-origin: 0% 100%;
  -webkit-animation: jNRJGA 125ms 250ms cubic-bezier(.4,.0,.23,1) forwards;
  animation: jNRJGA 125ms 250ms cubic-bezier(.4,.0,.23,1) forwards;
}

<div
  className="c0"
>
  <input
    id="id"
    primary={true}
    theme={
      Object {
        "dimensions": Object {
          "height": Object {
            "large": "40px",
            "medium": "30px",
            "small": "20px",
          },
          "icon": Object {
            "large": "24px",
            "medium": "16px",
            "small": "8px",
          },
        },
        "font": Object {
          "family": "Roboto, sans-serif",
          "lineHeight": 18,
          "size": "16px",
          "weight": 200,
        },
        "h1": Object {
          "fontSize": "28px",
          "fontWeight": 600,
          "lineHeight": "30px",
        },
        "h2": Object {
          "fontSize": "24px",
          "fontWeight": 400,
          "lineHeight": "28px",
        },
        "h3": Object {
          "fontSize": "20px",
          "fontWeight": 300,
          "lineHeight": "24px",
        },
        "h4": Object {
          "fontSize": "18px",
          "fontWeight": 300,
          "lineHeight": "20px",
        },
        "h5": Object {
          "fontSize": "14px",
          "fontWeight": 300,
          "lineHeight": "16px",
        },
        "h6": Object {
          "fontSize": "12px",
          "fontWeight": 300,
          "lineHeight": "14px",
        },
        "media": Object {
          "lg": "(min-width: 1280px)",
          "md": "(min-width: 980px)",
          "sm": "(min-width: 736px)",
          "xl": "(min-width: 1690px)",
          "xs": "(min-width: 480px)",
        },
        "palette": Object {
          "default": "#ffffff",
          "defaultBorder": "#bababa",
          "defaultDisabled": "#ededed",
          "defaultHover": "#f5f7f7",
          "defaultSelected": "#ededed",
          "defaultText": "#4c4c4c",
          "defaultTextDisabled": "#cecece",
          "disabled": "lightgray",
          "primary": "#1a5a99",
          "primaryBorder": "#bababa",
          "primaryDisabled": "#9ebbd7",
          "primaryHover": "#164872",
          "primarySelected": "#d8d8d8",
          "primaryText": "#ffffff",
          "primaryTextDisabled": "#c6d8e7",
          "secondary": "#5dcebb",
          "secondaryBorder": "#bababa",
          "secondaryDisabled": "#9ad1ec",
          "secondaryHover": "#217bad",
          "secondarySelected": "#bababa",
          "secondaryText": "#195382",
          "secondaryTextDisabled": "#cae5f4",
          "shadow": "black",
        },
        "radius": "3px",
        "spacing": Object {
          "borderSize": 1,
          "margin": "8px",
          "padding": Object {
            "bottom": "8px",
            "default": "8px",
            "left": "10px",
            "leftRight": "15px",
            "right": "10px",
            "top": "8px",
            "topBottom": "10px",
          },
          "shadowSize": 1,
        },
        "transitions": Object {
          "ease": ".25,.1,.25,1",
          "easeIn": ".42,0,1,1",
          "easeInOut": ".42,0,.58,1",
          "easeOut": "0,0,.58,1",
          "linear": "0,0,1,1",
        },
      }
    }
    type="checkbox"
  />
  <label
    htmlFor="id"
  >
    <span />
  </label>
</div>
`;

exports[`Checkbox should render a Checkbox 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0.5em;
}

.c0 * {
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 > input[type=checkbox] {
  visibility: hidden;
}

.c0 > input[type=checkbox] + label {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: .6em 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  -webkit-transition: color 250ms cubic-bezier(.4,.0,.23,1);
  transition: color 250ms cubic-bezier(.4,.0,.23,1);
}

.c0 > input[type=checkbox] + label > span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 1em;
  width: 20px;
  height: 20px;
  background: transparent;
  border: 2px solid #bababa;
  border-radius: 2px;
  cursor: pointer;
  -webkit-transition: all 250ms cubic-bezier(.4,.0,.23,1);
  transition: all 250ms cubic-bezier(.4,.0,.23,1);
}

.c0 > input[type=checkbox] + label:hover,
.c0 input[type=checkbox]:focus + label {
  color: #fff;
}

.c0 > input[type=checkbox] + label:hover > span,
.c0 input[type=checkbox]:focus + label > span {
  background: rgba(255,255,255,.1);
}

.c0 > input[type=checkbox]:checked + label > span {
  border: .6em solid #1a5a99;
  -webkit-animation: erhqVB 200ms cubic-bezier(.4,.0,.23,1);
  animation: erhqVB 200ms cubic-bezier(.4,.0,.23,1);
}

.c0 > input[type=checkbox]:checked + label > span:before {
  content: '';
  position: absolute;
  top: .5em;
  left: .25em;
  border-right: 3px solid transparent;
  border-bottom: 3px solid transparent;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-transform-origin: 0% 100%;
  -ms-transform-origin: 0% 100%;
  transform-origin: 0% 100%;
  -webkit-animation: jNRJGA 125ms 250ms cubic-bezier(.4,.0,.23,1) forwards;
  animation: jNRJGA 125ms 250ms cubic-bezier(.4,.0,.23,1) forwards;
}

<div
  className="c0"
>
  <input
    id="id"
    primary={true}
    theme={
      Object {
        "dimensions": Object {
          "height": Object {
            "large": "40px",
            "medium": "30px",
            "small": "20px",
          },
          "icon": Object {
            "large": "24px",
            "medium": "16px",
            "small": "8px",
          },
        },
        "font": Object {
          "family": "Roboto, sans-serif",
          "lineHeight": 18,
          "size": "16px",
          "weight": 200,
        },
        "h1": Object {
          "fontSize": "28px",
          "fontWeight": 600,
          "lineHeight": "30px",
        },
        "h2": Object {
          "fontSize": "24px",
          "fontWeight": 400,
          "lineHeight": "28px",
        },
        "h3": Object {
          "fontSize": "20px",
          "fontWeight": 300,
          "lineHeight": "24px",
        },
        "h4": Object {
          "fontSize": "18px",
          "fontWeight": 300,
          "lineHeight": "20px",
        },
        "h5": Object {
          "fontSize": "14px",
          "fontWeight": 300,
          "lineHeight": "16px",
        },
        "h6": Object {
          "fontSize": "12px",
          "fontWeight": 300,
          "lineHeight": "14px",
        },
        "media": Object {
          "lg": "(min-width: 1280px)",
          "md": "(min-width: 980px)",
          "sm": "(min-width: 736px)",
          "xl": "(min-width: 1690px)",
          "xs": "(min-width: 480px)",
        },
        "palette": Object {
          "default": "#ffffff",
          "defaultBorder": "#bababa",
          "defaultDisabled": "#ededed",
          "defaultHover": "#f5f7f7",
          "defaultSelected": "#ededed",
          "defaultText": "#4c4c4c",
          "defaultTextDisabled": "#cecece",
          "disabled": "lightgray",
          "primary": "#1a5a99",
          "primaryBorder": "#bababa",
          "primaryDisabled": "#9ebbd7",
          "primaryHover": "#164872",
          "primarySelected": "#d8d8d8",
          "primaryText": "#ffffff",
          "primaryTextDisabled": "#c6d8e7",
          "secondary": "#5dcebb",
          "secondaryBorder": "#bababa",
          "secondaryDisabled": "#9ad1ec",
          "secondaryHover": "#217bad",
          "secondarySelected": "#bababa",
          "secondaryText": "#195382",
          "secondaryTextDisabled": "#cae5f4",
          "shadow": "black",
        },
        "radius": "3px",
        "spacing": Object {
          "borderSize": 1,
          "margin": "8px",
          "padding": Object {
            "bottom": "8px",
            "default": "8px",
            "left": "10px",
            "leftRight": "15px",
            "right": "10px",
            "top": "8px",
            "topBottom": "10px",
          },
          "shadowSize": 1,
        },
        "transitions": Object {
          "ease": ".25,.1,.25,1",
          "easeIn": ".42,0,1,1",
          "easeInOut": ".42,0,.58,1",
          "easeOut": "0,0,.58,1",
          "linear": "0,0,1,1",
        },
      }
    }
    type="checkbox"
  />
  <label
    htmlFor="id"
  >
    <span />
  </label>
</div>
`;
